### DISCUSSION TITLE
Documentation of the buffer method

### DISCUSSION THREAD

**Discussion by RamziBrini** [locationtech/jts#917] [Q&A] [answered]
Hi,

I am using the buffer(0) method to get rid of self-intersecting geometries(following articles of @dr-jts). So far, it fixes my problem but the method documentation seem to be misleading:

org.locationtech.jts.geom.Geometry#buffer(double)

"The buffer operation always returns a polygonal result. The negative or zero-distance buffer of lines and points is always an empty Polygon. This is also the result for the buffers of degenerate (zero-area) polygons."

According to my observation this is not true. Could someone confirm?

Best Regards

**Comment by dr-jts** [comment, marked as answer]
If you think you've found an example which contradicts the documentation, please post it.

Also, the relatively new [`GeometryFixer`](https://locationtech.github.io/jts/javadoc/org/locationtech/jts/geom/util/GeometryFixer.html) class is a better way of repairing invalid geometry.

**Reply by RamziBrini** [reply]
Thank you for your recommendation. I guess GeometryFixer will be my first choice to get rid of self intersecting geometries. I got stuck to your article from December 23, 2020 about "Fixing buffer for fixing polygons" without noticing an even more sophisticated way that you described in your article from May 3rd, 2021.

Back to my original request, I think I misunderstood the sentence "...zero-distance buffer of lines and points is always an empty Polygon" 

Thanks
